01_network_sysapi/Messenger/MessengerNetworking/src/tcp_client.cpp

```diff
@@ -0,0 +1,53 @@
+#include "MessengerNetworking/tcp_client.h"
+void tcpClient::bind(){
+	hint.sin_family = AF_INET;
+	hint.sin_port = htons(conPort);
+    auto validate = inet_addr(ipAdress.c_str());
+	
+    if(validate == INADDR_NONE)
+            throw new std::exception("IP validation has failed.");
+	hint.sin_addr.S_un.S_addr = validate;
+}
+void tcpClient::connect(){
+    initializeWinsock();
+    createSocket();
+    bind();
+    int connResult = ::connect(sock, (sockaddr*)&hint, sizeof(hint));
+	if (connResult == SOCKET_ERROR)
+		throw new std::exception(("Can't connect to server, Err #" + std::to_string(WSAGetLastError())).c_str());
+    char buf[4096];
+	std::string userInput="";
+	std::string msgRecv = "";
+    do
+	{
+		std::cout << "> ";
```

Comment:
Chat functionality (messages input/output) tightly mixed with network communication code. These are different parts of system and should be divided. Pay attention to this in future projects. Entities should be single responsible.